src/main/java/redis/clients/jedis/UnifiedJedis.java

```diff
@@ -4822,8 +4822,11 @@ public Object tFunctionCallAsync(String library, String function, List<String> k
   public PipelineBase pipelined() {
     if (provider == null) {
       throw new IllegalStateException("It is not allowed to create Pipeline from this " + getClass());
+    } else if (provider instanceof MultiClusterPooledConnectionProvider) {
+      return new MultiClusterFailoverPipeline((MultiClusterPooledConnectionProvider) provider);
+    } else {
+      return new Pipeline(provider.getConnection(), true);
     }
-    return new Pipeline(provider.getConnection(), true);
   }
 
   public Transaction multi() {
```

Comment:
@sazzad16 See my comment in https://github.com/redis/jedis/issues/3568#issuecomment-1750799211.

This is also a problems for multi(), subscribe(), and any methods that use JedisCommandIterationBase. Shouldn't we fix these as well?

Reply:
Current plan: Those will be in a different issue, and estimated there.

Reply:
#3602 contains multi and supersedes this subscribe